includes/Modules/AdSense.php

```diff
@@ -835,6 +849,32 @@ private function register_tag() {
 		}
 	}
 
+	/**
+	 * Checks if the module tag is found in the provided content.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $content Content to search for the tags.
+	 * @return bool TRUE if tag is found, FALSE if not.
+	 */
+	public function has_placed_tag_in_content( $content ) {
+		$tag_matchers = ( new Tag_Matchers() )->regex_matchers();
+
+		$search_string = 'Google AdSense snippet added by Site Kit';
+
+		if ( strpos( $content, $search_string ) !== false ) {
+			return Module_Tag_Matchers::TAG_EXISTS_WITH_COMMENTS;
+		} else {
+			foreach ( $tag_matchers as $pattern ) {
+				if ( preg_match( $pattern, $content ) ) {
+					return Module_Tag_Matchers::TAG_EXISTS;
+				}
+			}
+		}
+
+		return Module_Tag_Matchers::NO_TAG_FOUND;
+	}
```

Comment:
This seems like it would be the same in all classes and so we could extract it to a trait. We'd only need the class to have a method to provide the `Tag_Matchers` instance.

As mentioned in the issue though, using AMP removes all the comments so that would never match.

Reply:
It was moved between Trait and required method during CR. Added it as Trait, thanks